architecture/replication-layer.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Architecture&#58; Replication Layer
+summary: 
+toc: false
+---
+
+The Replication Layer of CockroachDB's architecture copies data between nodes and implements and ensures consistency between these copies by implementing our consensus algorithm.
+
+{{site.data.alerts.callout_info}}If you haven't already, we recommend reading the <a href="overview.html">Architecture Overivew</a>.{{site.data.alerts.end}}
+
+<div id="toc"></div>
+
+## Overview
+
+High availability requires that your database can tolerate nodes going offline without interrupting service to your application. This means replicating data between nodes to ensure the data remains accessible.
+
+Ensuring consistency with nodes offline, though, is a challenge many databases fail. To solve this problem, CockroachDB uses consistent replication, which uses a consensus algorithm to require a quorum of replicas must agree on any changes to the range before they're considered committed. Because 3 is the smallest number that can achieve quorum (i.e., 2 out of 3), CockroachDB's high availability (known as Multi-Active Availability) requires 3 nodes.
+
+The number of failures that can be tolerated is equal to *(Replication factor - 1)/2*. For example, with 3x replication, one failure can be tolerated; with 5x replication, two failures, and so on. You can control the replication factor per-table in your cluster's [Replication Zones](stable/configure-replication-zones.html).
+
+When failures happen, though, CockroachDB automatically realizes nodes have stopped responding and work to redistribute your data to continue maximizing survivability. This process also works the other way around: when new nodes join your cluster, data automatically rebalances onto it, ensuring your load is evenly distributed.
+
+### Interactions with Other Layers
+
+In relationship to other layers in CockroachDB, the Distribution Layer:
+
+- Receives requests from and sends responses to the Distribution Layer.
+- Writes accepted requests to the Storage Layer.
+
+## Components
+
+### Raft
+
+Raft is a consensus protocol––an algorithm which makes sure that your data is safely stored on multiple machines, and that those machines agree on the current state even if some of them are temporarily disconnected.
+
+Raft organizes all nodes that contain a replica of a range into a group––unsurprisingly called a Raft Group.
```

Comment:
I think it might be good to combine this and the following paragraph, with a transition introducing the concept of leader/follower, e.g.:

> Raft organizes all nodes that contain a replica of a range into a group--unsurprisingly call a Raft Group. Each replica in a Raft Group is either a "leader" or a "follower". The leader, which is elected by Raft and long-lived, coordinates all writes to the Raft Group. It heartbeats followers periodically and keeps their logs replicated...